bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/CommandTag.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.icalendar.internal.logic;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.HSBType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.PlayPauseType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.RewindFastforwardType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.TypeParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a class that implements a Command Tag that may be embedded in an
+ * Event Description. Valid Tags must follow one of the following forms..
+ *
+ * BEGIN:<itemName>:<targetState>
+ * BEGIN:<itemName>:<targetState>:<authorizationCode>
+ * END:<itemName>:<targetState> END:<itemName>:<targetState>:<authorizationCode>
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class CommandTag {
+    private @Nullable String itemName;
+    private @Nullable String targetState;
+    private String fullTag;
+    private @Nullable CommandTagType tagType;
+    private boolean isValid = false;
+    private @Nullable String authorizationCode;
+
+    private final Logger logger = LoggerFactory.getLogger(CommandTag.class);
+
+    private static final List<Class<? extends Command>> percentCommandType = Arrays.asList(PercentType.class);
+
+    private static final List<Class<? extends Command>> otherCommandTypes = Arrays.asList(QuantityType.class,
+            OnOffType.class, OpenClosedType.class, UpDownType.class, HSBType.class, PlayPauseType.class,
+            RewindFastforwardType.class, StringType.class);
+
+    public String getItemName() {
+        return itemName != null ? ((String) itemName) : "";
```

Comment:
It wasn't really intended for you to cast your way out of null checks. The warning you were encountering here was mostly to remind you that your null check wasn't thread safe. To properly solve it you could either make that field final or use a temporary variable like so:
```suggestion
        String itemName = this.itemName;
        return itemName != null ? ((String) itemName) : "";
```

Reply:
Oh, yes strange why i came to the idea to do it that way as i've done it the right way through all the other code...